modules/database/src/ioc/db/dbAccess.c

```diff
@@ -853,7 +853,7 @@ static long getLinkValue(DBADDR *paddr, short dbrType,
     {
         const char *rtnString = dbGetString(&dbEntry);
 
-        strncpy(pbuf, rtnString, maxlen-1);
+        strncpy(pbuf, rtnString, maxlen);
```

Comment:
Interesting, I can't imagine which warning this would fix. What happened here, which compiler complained? I can't reproduce it with gcc 10.3.

Reply:
Maybe the compiler thinks that `maxlen-1` may become negative (if `*nRequest == 0`) but that is actually taken care of in line 830. And it would be a problem in like 857 too.

Reply:
I am a bit surprised that the  compiler does not complain about line 890. It looks similar to me.